### PR TITLE
DWR-845, changes for district/school groups

### PR DESCRIPTION
**Pull request by malaffoon**
Initial changes to admin for district/school groups. Most of the change is from moving code into common (to share with reporting). I'll try to call out the interesting bits.

I will update this to bump common dependency when that review is blessed.

NOTE: there are more code changes before all this is working. All the queries that use Permission need to be changed to use the extended PermissionScope.

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/admin/Application.java [hunk 1]
```diff
@@ -24,7 +25,8 @@
         EndpointAutoConfiguration.class,
         StatusConfiguration.class,
         YamlPropertiesConfigurator.class,
-        DataSourceConfiguration.class
+        DataSourceConfiguration.class,
+        JdbcOrganizationRepository.class
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/Application.java:29]: This is a @Component in RDW_Common. This is the least impactful way to force it into this project; component scanning caused some odd behavior with the controllers.

File: webapp/src/main/java/org/opentestsystem/rdw/admin/security/auth/DefaultAuthorizationService.java [hunk 2]
```diff
@@ -3,13 +3,13 @@
 import com.google.common.collect.ImmutableSet;
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/security/auth/DefaultAuthorizationService.java:3]: This class is where most of the real changes are. 

Yeah, i could try to pull this whole class into RDW_Common but it started to get messy. Maybe next go-around.

File: webapp/src/main/resources/application.sql.yml [hunk 3]
```diff
@@ -3,12 +3,24 @@ sql:
     findWarehouseImports: >-
```
Comment (malaffoon) [inline webapp/src/main/resources/application.sql.yml]: oops, these shouldn't be here any more

File: webapp/src/test/java/org/opentestsystem/rdw/admin/security/auth/DefaultAuthorizationServiceTest.java [hunk 4]
```diff
@@ -77,36 +73,37 @@ public void getGrantsShouldOnlyIncludeGrantsForTheSpecifiedState() throws Except
...
 
     @Test
     public void getPermissionsShouldCorrectlyDerivePermissionsFromGrants() throws Exception {
         final List<Grant> grants = ImmutableList.of(
-                Grant.builder().entityId("a").role("a").entityLevel(STATE).stateId("a").build(),
-                Grant.builder().entityId("a").role("b").entityLevel(DISTRICT).stateId("a").districtId("a").build(),
-                Grant.builder().entityId("a").role("c").entityLevel(INSTITUTION).stateId("a").districtId("a").institutionId("a").build(),
-                Grant.builder().entityId("a").role("d").entityLevel(DISTRICT).stateId("a").districtId("a").build(),
-                Grant.builder().entityId("a").role("d").entityLevel(INSTITUTION).stateId("a").districtId("a").institutionId("a").build()
+                Grant.fromString("|a|a|STATE|||||a||"),
```
Comment (malaffoon) [inline webapp/src/test/java/org/opentestsystem/rdw/admin/security/auth/DefaultAuthorizationServiceTest.java:83]: i didn't retain the Grant builder so i had to convert these to tenancy chains

File: common/src/test/java/org/opentestsystem/rdw/admin/common/test/support/PermissionScopes.java [hunk 5]
```diff
@@ -16,11 +16,11 @@ public static PermissionScope statewide() {
     }
 
     public static PermissionScope districts(final Long... ids) {
-        return new PermissionScope(Arrays.asList(ids), null);
+        return new PermissionScope(Arrays.asList(ids), null, null, null);
```
Comment (esotrope) [inline common/src/test/java/org/opentestsystem/rdw/admin/common/test/support/PermissionScopes.java:19]: uh oh i think we broke our self-imposed 3 param limit and need a builder now : p

**Review by esotrope — APPROVED** [review]

File: common/src/test/java/org/opentestsystem/rdw/admin/common/test/support/PermissionScopes.java (hunk 5 above)
Comment (malaffoon) [reply common/src/test/java/org/opentestsystem/rdw/admin/common/test/support/PermissionScopes.java:19]: Dang, i was hoping nobody would catch that.

Comment (esotrope) [reply common/src/test/java/org/opentestsystem/rdw/admin/common/test/support/PermissionScopes.java:19]: Hard to miss with all those consecutive nulls. I thought you were instantiating an array x J

Comment (jtreuting) [reply common/src/test/java/org/opentestsystem/rdw/admin/common/test/support/PermissionScopes.java:19]: I believe I said 5 to start and someone (.... @malaffoon ) thought 3 was better

File: webapp/src/main/resources/application.sql.yml [hunk 6]
```diff
@@ -3,12 +3,6 @@ sql:
...
       SELECT i.id, i.digest FROM import i limit 10;
 
-  district:
-    findAllIds: >-
```
Comment (jtreuting) [inline webapp/src/main/resources/application.sql.yml:7]: where did the findAllIds go?  not used or moved to the common application.sql.yml?

Comment (malaffoon) [reply webapp/src/main/resources/application.sql.yml:7]: moved into common

**Pull request merged**